src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -0,0 +1,184 @@
+﻿using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    /// <summary>
+    /// Parse a project.json, and add additional dependencies, then write a out new project.json.
+    /// </summary>
+    public class AddDependenciesToProjectJson : Task
+    {
+        private Regex VersionStructureRegex = new Regex(@"(\d+\.\d+\.\d+)-((beta|rc2|rc3)?-?(\d+))");
```

Comment:
Not blocker right now but I don't like that these Regex make the assumption about our prerelease IDs. These should probably be task parameters similar to how we do for the package version validation tasks.